### ISSUE TITLE
Failed to run install_centos_stable_deps()

### ISSUE THREAD

**Issue by Blacktiger** [saltstack/salt-bootstrap#628]
Relevant log information.

```
 *  INFO: sh install_salt.sh -- Version 2015.07.17

 *  INFO: System Information:
 *  INFO:   CPU:          GenuineIntel
 *  INFO:   CPU Arch:     x86_64
 *  INFO:   OS Name:      Linux
 *  INFO:   OS Version:   2.6.32-504.8.1.el6.x86_64
 *  INFO:   Distribution: CentOS 6.6

 * DEBUG: Binaries will be searched using the following $PATH: /usr/lib64/qt-3.3/bin:/opt/java/bin:/sbin:/bin:/usr/sbin:/usr/bin:/usr/local/sbin:/usr/local/bin:/home/martzm/bin
 *  INFO: Installing minion
 * DEBUG: install_centos_6_stable_deps not found....
 * DEBUG: install_centos_6_6_stable_deps not found....
 * DEBUG: install_centos_6_deps not found....
 * DEBUG: install_centos_6_6_deps not found....
 *  INFO: Found function install_centos_stable_deps
 * DEBUG: DEPS_INSTALL_FUNC=install_centos_stable_deps
 * DEBUG: CONFIG_SALT_FUNC=null
 * DEBUG: PRESEED_MASTER_FUNC=null
 * DEBUG: install_centos_6_stable not found....
 * DEBUG: install_centos_6_6_stable not found....
 *  INFO: Found function install_centos_stable
 * DEBUG: INSTALL_FUNC=install_centos_stable
 * DEBUG: install_centos_6_stable_post not found....
 * DEBUG: install_centos_6_6_stable_post not found....
 * DEBUG: install_centos_6_post not found....
 * DEBUG: install_centos_6_6_post not found....
 *  INFO: Found function install_centos_stable_post
 * DEBUG: POST_INSTALL_FUNC=install_centos_stable_post
 * DEBUG: install_centos_6_stable_restart_daemons not found....
 * DEBUG: install_centos_6_6_stable_restart_daemons not found....
 * DEBUG: install_centos_6_restart_daemons not found....
 * DEBUG: install_centos_6_6_restart_daemons not found....
 * DEBUG: install_centos_stable_restart_daemons not found....
 *  INFO: Found function install_centos_restart_daemons
 * DEBUG: STARTDAEMONS_INSTALL_FUNC=install_centos_restart_daemons
 * DEBUG: daemons_running_centos_6_stable not found....
 * DEBUG: daemons_running_centos_6_6_stable not found....
 * DEBUG: daemons_running_centos_6 not found....
 * DEBUG: daemons_running_centos_6_6 not found....
 * DEBUG: daemons_running_centos_stable not found....
 * DEBUG: daemons_running_centos not found....
 *  INFO: Found function daemons_running
 * DEBUG: DAEMONS_RUNNING_FUNC=daemons_running
 * DEBUG: install_centos_6_stable_check_services not found....
 * DEBUG: install_centos_6_6_stable_check_services not found....
 * DEBUG: install_centos_6_check_services not found....
 * DEBUG: install_centos_6_6_check_services not found....
 * DEBUG: install_centos_stable_check_services not found....
 *  INFO: Found function install_centos_check_services
 * DEBUG: CHECK_SERVICES_FUNC=install_centos_check_services
 *  INFO: Running install_centos_stable_deps()
 *  INFO: Adding SaltStack's COPR repository
Loaded plugins: fastestmirror, priorities, security, versionlock


Error getting repository data for epel, repository not found
 * ERROR: Failed to run install_centos_stable_deps()!!!
 * DEBUG: Removing the logging pipe /tmp/bootstrap-salt.logpipe
 * DEBUG: Killing logging pipe tee's with pid(s): 9883
```

**Comment by astro44** [comment]
also got same error on both centos 6.6-x86_64and 7.1-x86_64

**Comment by ninjada** [comment]
experiencing on centos 6.6 with my vagrant box when running salt highstate

shouldnt be OS specific. looks to be a repository issue, the http://copr.fedoraproject.org link is not resolving atm..

i believe this happens occasionally. is there a decent workaround? like lowering the timeout or something so it will skip things?

**Comment by sambanshee** [comment]
Same issue here. CentOS 6.6 
Manually updated to latest version of script - 2015.07.17 - still errors:

```
           Salt: 2015.5.2
         Python: 2.6.6 (r266:84292, Jan 22 2014, 09:42:36)
         Jinja2: 2.2.1
       M2Crypto: 0.20.2
 msgpack-python: 0.4.6
   msgpack-pure: Not Installed
       pycrypto: 2.0.1
        libnacl: Not Installed
         PyYAML: 3.10
          ioflo: Not Installed
          PyZMQ: 14.3.1
           RAET: Not Installed
            ZMQ: 3.2.5
           Mako: Not Installed
```

```
 * ERROR: Failed to run install_centos_stable_deps()!!!
12:13:43            Process Process-5:
12:13:43            Traceback (most recent call last):
12:13:43              File "/usr/lib64/python2.6/multiprocessing/process.py", line 232, in _bootstrap
12:13:43                self.run()
12:13:43              File "/usr/lib64/python2.6/multiprocessing/process.py", line 88, in run
12:13:43                self._target(*self._args, **self._kwargs)
12:13:43              File "/usr/lib/python2.6/site-packages/salt/cloud/__init__.py", line 1241, in create
12:13:43                output = self.clouds[func](vm_)
12:13:43              File "/usr/lib/python2.6/site-packages/salt/cloud/clouds/vsphere.py", line 262, in create
12:13:43                deploy_kwargs = _deploy(vm_)
12:13:43              File "/usr/lib/python2.6/site-packages/salt/cloud/clouds/vsphere.py", line 421, in _deploy
12:13:43                deployed = salt.utils.cloud.deploy_script(**deploy_kwargs)
12:13:43              File "/usr/lib/python2.6/site-packages/salt/utils/cloud.py", line 1299, in deploy_script
12:13:43                if root_cmd(deploy_command, tty, sudo, **ssh_kwargs) != 0:
12:13:43              File "/usr/lib/python2.6/site-packages/salt/utils/cloud.py", line 1795, in root_cmd
12:13:43                retcode = _exec_ssh_cmd(cmd, allow_failure=allow_failure, **kwargs)
12:13:43              File "/usr/lib/python2.6/site-packages/salt/utils/cloud.py", line 1476, in _exec_ssh_cmd
12:13:43                cmd, proc.exitstatus
12:13:43            SaltCloudSystemExit: Command 'ssh -oStrictHostKeyChecking=no -oUserKnownHostsFile=/dev/null -oControlPath=none -p 22 root@172.17.10.63 \'/tmp/.saltcloud-48211b93-0336-4c40-9ea5-4505c7e7c564/deploy.sh -c \'"\'"\'/tmp/.saltcloud-48211b93-0336-4c40-9ea5-4505c7e7c564\'"\'"\'\'' failed. Exit code: 1

```

**Comment by s0undt3ch** [comment]
This seems to have been a temporary resolution failure/downtime of the COPR repository. It's apparently back up. Could you please try again?

**Comment by sambanshee** [comment]
@s0undt3ch while COPR is back online and bootstrap works again I still believe that the error could give more info (like copr.fedoraproject... is unreacheable). 

Would be also nice to have an option to increase a timeout or skip this step if possible as it took a while for COPR to be back.

**Comment by s0undt3ch** [comment]
@sambanshee We cannot skip the COPR step since it provides the required package versions to install Salt.

About the actual error:

```
 *  INFO: Adding SaltStack's COPR repository
Loaded plugins: fastestmirror, priorities, security, versionlock


Error getting repository data for epel, repository not found
 * ERROR: Failed to run install_centos_stable_deps()!!
```

The above error is clarifying, however your output is from a salt-cloud execution which might be more confusing but I believe that the error message I pasted above can also be seen on the salt-cloud execution output.

Closing as fixed since this wasn't an issue with the bootstrap script.
